### PR TITLE
Module5

### PR DESCRIPTION
**Pull request by BatiushkovNikita**

### CODE REVIEW

File: auth/auth-strategy.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+const twitter = () => {
+    passport.use(new TwitterStrategy({
+            consumerKey: 'TWITTER_CONSUMER_KEY',
+            consumerSecret: 'TWITTER_CONSUMER_SECRET',
+            callbackURL: "http://127.0.0.1:3000/auth/twitter/callback"
+        },
+        (token, tokenSecret, profile, cb) => {
+            User.findOrCreate({twitterId: profile.id}, function (err, user) {
```
Comment (vinfinit) [inline auth/auth-strategy.js]: `User.findOrCreate({twitterId: profile.id}, cb);`

File: auth/auth-strategy.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+const googleOAuth = () => {
+    passport.use(new GoogleStrategy({
+            clientID: 'GOOGLE_CLIENT_ID',
+            clientSecret: 'GOOGLE_CLIENT_SECRET',
+            callbackURL: "http://yourdormain:3000/auth/google/callback",
+            passReqToCallback: true
+        },
+        (request, accessToken, refreshToken, profile, done) => {
+            User.findOrCreate({googleId: profile.id}, function (err, user) {
```
Comment (vinfinit) [inline auth/auth-strategy.js]: `User.findOrCreate({googleId: profile.id}, done);`

File: auth/auth-strategy.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+const GoogleStrategy = passportGoogleOAuth.Strategy;
+
+const local = () => {
+    passport.use(new LocalStrategy({
+            passwordField: 'email'
+        },
+        (username, password, done) => {
+            let user = usersData.find((user) => {
+                return user.email === password && user.username === username;
+            });
```
Comment (vinfinit) [inline auth/auth-strategy.js]:
```javascript
let user = usersData.find(user => (
    user.email === password && user.username === username
));
```

File: auth/auth-strategy.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+const facebook = () => {
+    passport.use(new FacebookStrategy({
+            clientID: 'FACEBOOK_APP_ID',
+            clientSecret: 'FACEBOOK_APP_SECRET',
+            callbackURL: "http://localhost:3000/auth/facebook/callback"
+        },
+        (accessToken, refreshToken, profile, cb) => {
+            User.findOrCreate({facebookId: profile.id}, function (err, user) {
```
Comment (vinfinit) [inline auth/auth-strategy.js]: `User.findOrCreate({facebookId: profile.id}, cb);`

Comment (BatiushkovNikita) [reply auth/auth-strategy.js]: fixed

File: auth/auth-strategy.js (hunk 3 above)
Comment (BatiushkovNikita) [reply auth/auth-strategy.js]: fixed

File: auth/auth-strategy.js (hunk 2 above)
Comment (BatiushkovNikita) [reply auth/auth-strategy.js]: fixed

File: auth/auth-strategy.js (hunk 1 above)
Comment (BatiushkovNikita) [reply auth/auth-strategy.js]: fixed

**Pull request merged**